src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseExportImportResource.java

```diff
@@ -251,10 +253,15 @@ public ResponseEntity<ProgrammingExercise> importProgrammingExercise(@PathVariab
                     .body(importedProgrammingExercise);
 
         }
-        catch (Exception ex) {
-            log.error(ex.getMessage(), ex);
-            return ResponseEntity.internalServerError()
-                    .headers(HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, "importExerciseTriggerPlanFail", "Unable to import programming exercise")).build();
+        catch (Exception exception) {
+            log.error(exception.getMessage(), exception);
+
+            boolean isExceptionWithTranslationKeys = exception instanceof HttpStatusException;
+            if (isExceptionWithTranslationKeys) {
+                throw exception;
+            }
```

Comment:
The issue that caused no alert to be seen on the client side was caused by the `BadRequestAlertException` being caught in this block